### PR TITLE
MM-20705 Support Kubernetes version 1.16 in mattermost-team-edition

### PR DESCRIPTION
**Pull request by bpietraga**
Relates to https://github.com/mattermost/mattermost-server/issues/13227

# What it does

This PR only address the `mattermost-team-edition`

- Updates Ingress `apiVersion` from `extensions/v1beta1` to `networking.k8s.io/v1beta1`  based on [kubernetes/kubernetes/pull/70672](https://github.com/kubernetes/kubernetes/pull/70672)
- After change bumps minimal required Kubernetes version to `1.14+`
- Fixes syntax highlighting in `README.md` markdown.
- Add *Local development* section to readme, which might help people developing locally. This can be removed if is not necessary.

# Why it is important

- Support new versions of Kubernetes

# Notes to reviewer

The PR code was tested on k8 versions: 
- `1.15.2`
- `1.16.2`

I've didn't change comment  `volume.beta.kubernetes.io/storage-class` in 
https://github.com/bpietraga/mattermost-helm/blob/support_kubernetes_1_16/charts/mattermost-team-edition/values.yaml#L27
The `volume.beta.kubernetes.io/storage-class` is being deprecated in favor `StorageClassName`, but it is just example comment and  `volume.beta.kubernetes.io/storage-class` still works

### CODE REVIEW

**Comment by bpietraga** [comment]
I've filled CLA form, looks like the check is still failing

File: charts/mattermost-team-edition/templates/ingress.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 {{ if $ingress.enabled }}
 {{ $serviceName := include "mattermost-team-edition.name" . }}
 {{ $servicePort := .Values.service.externalPort }}
-apiVersion: extensions/v1beta1
+apiVersion: networking.k8s.io/v1beta1
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply charts/mattermost-team-edition/templates/ingress.yaml]: Clarification, reverting this change would still make this helm chart support k8s 1.16.

Comment (consideRatio) [inline charts/mattermost-team-edition/templates/ingress.yaml]:
```suggestion
apiVersion: extensions/v1beta1
```

[the remark this answers is not in the document]
Comment (cpanato) [reply charts/mattermost-team-edition/templates/ingress.yaml]: agree

[the remark this answers is not in the document]
Comment (bpietraga) [reply charts/mattermost-team-edition/templates/ingress.yaml]: Commited in https://github.com/mattermost/mattermost-helm/pull/126/commits/0cdc9dcbf2fab1cc34ba24ef0e8dbdc76366f24f

Thanks @consideRatio @cpanato  :+1:

**Comment by cpanato** [comment]
@bpietraga did you test an upgrade from the current chart to this new version? this is not going to break any current deployments?

File: charts/mattermost-team-edition/README.md [hunk 2]
```diff
@@ -18,7 +18,9 @@ cluster using the [Helm](https://helm.sh) package manager.
 
 ## Prerequisites
 
-- Kubernetes 1.8+ with Beta APIs enabled
+- Kubernetes 1.14+ with Beta APIs enabled
```
Comment (consideRatio) [inline charts/mattermost-team-edition/README.md]: I think this is more accurate based on the changes in this PR at least.
```suggestion
- Kubernetes 1.9+ with Beta APIs enabled
```

**Comment by bpietraga** [comment]
Thanks @consideRatio for code suggestions!

**Comment by bpietraga** [comment]
> @bpietraga did you test an upgrade from the current chart to this new version? this is not going to break any current deployments?

I will check double check it tomorrows morning.  As the changes has been applied. But it looks like no chart has been changed at the end and only `README` and chart version is in the diff.

**Comment by jasonblais** [comment]
Hey @bpietraga, just touching base to see if you have any questions we can help with? :)

**Comment by cpanato** [comment]
@bpietraga doing the review for this PR I notice there is no changes in the manifests, so that means this is good for k8s v1.16+?

**Comment by bpietraga** [comment]
> @bpietraga doing the review for this PR I notice there are no changes in the manifests, so that means this is good for k8s v1.16+?

Yes after checking the migrations, looks like the charts go even on the `1.17`. Previous commits would target the to be deprecated bits. But as the 1.17 is working fine i think the PR can be closed. 

Only things is the readme improvements which could be useful for newcomers. But this is topic for separate PR.

**Comment by jasonblais** [comment]
@cpanato / @stylianosrigas friendly reminder to help with reviews after the holidays :)

**Pull request merged**